build_library/catalyst_sdk.sh

```diff
@@ -7,8 +7,8 @@ source /tmp/toolchain_util.sh
 echo "Double checking everything is fresh and happy."
 run_merge -uDN --with-bdeps=y world
 
-echo "Setting the default Python interpreter to Python 2."
-eselect python set python2.7
+echo "Setting the default Python interpreter to Python 3."
+eselect python set python3.9
```

Comment:
Do you think we should have something like at the end of `update_chroot` as well? So that chroots continue working after python 2.7 is gone?
Also: any thoughts on `eselect python update`?

Reply:
> Do you think we should have something like at the end of `update_chroot` as well? So that chroots continue working after python 2.7 is gone?

This makes sense, I'll test it.

> Also: any thoughts on `eselect python update`?

Never heard of it before. :) But it's certainly nicer, as we wouldn't have to update the python version in script next time. I'll try it out.

Reply:
Trying to run update chroot in some older SDK:
```
$ VERSION=2021.12.03+dev-flatcar-master-4304
$ ./set_version --binhost https://bucket.release.flatcar-linux.net/flatcar-jenkins --dev-board --board-version "${VERSION}" --dev-sdk --sdk-version "${VERSION}"
$ ./update_chroot
…
Calculating dependencies... done!

!!! The following binary packages have been ignored due to non matching USE:

    =sys-apps/kmod-29 -python_targets_python3_6 python_targets_python3_9
    =sys-apps/util-linux-2.37.2-r1 python_targets_python3_9
    =app-misc/pax-utils-1.3.1 python_single_target_python3_6
    =dev-libs/libxml2-2.9.12-r5 python_targets_python3_9
    =dev-util/meson-0.57.2 -python_targets_python3_6 python_targets_python3_9
    =dev-util/meson-format-array-0 -python_targets_python3_6 python_targets_python3_9
    =dev-vcs/git-2.32.0-r1 python_single_target_python3_9
    =sys-libs/libcap-ng-0.8.2-r1 -python_targets_python3_6 python_targets_python3_9
    =sys-process/audit-3.0.6 -python_targets_python3_6 python_targets_python3_9
    =sys-fs/btrfs-progs-5.15.1 -python_single_target_python3_6 python_single_target_python3_9 -python_targets_python3_6

NOTE: The --binpkg-respect-use=n option will prevent emerge
      from ignoring these binary packages if possible.
      Using --binpkg-respect-use=y will silence this warning.

!!! The following binary packages have been ignored due to changed dependencies:

     sys-apps/coreutils-8.32-r1::portage-stable
     dev-util/ninja-1.10.2-r1::portage-stable
     dev-libs/jsoncpp-1.9.4::portage-stable
     coreos-base/hard-host-depends-0.0.1-r199::coreos

NOTE: The --binpkg-changed-deps=n option will prevent emerge
      from ignoring these binary packages if possible.
      Using --binpkg-changed-deps=y will silence this warning.

!!! The ebuild selected to satisfy "dev-libs/gobject-introspection" has unmet requirements.
- dev-libs/gobject-introspection-1.66.1::portage-stable USE="-doctool -gtk-doc -test" PYTHON_SINGLE_TARGET="-python3_6 (-python3_7) (-python3_8) (-python3_9)" PYTHON_TARGETS="python3_6 (-python3_7) (-python3_8) (-python3_9)"

  The following REQUIRED_USE flag constraints are unsatisfied:
    exactly-one-of ( python_single_target_python3_6 python_single_target_python3_7 python_single_target_python3_8 python_single_target_python3_9 )

  The above constraints are a subset of the following complete expression:
    exactly-one-of ( python_single_target_python3_6 python_single_target_python3_7 python_single_target_python3_8 python_single_target_python3_9 ) python_single_target_python3_6? ( python_targets_python3_6 ) python_single_target_python3_7? ( python_targets_python3_7 ) python_single_target_python3_8? ( python_targets_python3_8 ) python_single_target_python3_9? ( python_targets_python3_9 )

(dependency required by "coreos-devel/sdk-depends-0.0.1-r41::coreos" [installed])
(dependency required by "coreos-devel/sdk-depends" [argument])
ERROR   update_chroot: script called: update_chroot 
ERROR   update_chroot: Backtrace:  (most recent call is last)
ERROR   update_chroot:   file update_chroot, line 260, called: die_err_trap 'sudo -E ${EMERGE_CMD} "${EMERGE_FLAGS[@]}" coreos-devel/sdk-depends world' '1'
ERROR   update_chroot: 
ERROR   update_chroot: Command failed:
ERROR   update_chroot:   Command 'sudo -E ${EMERGE_CMD} "${EMERGE_FLAGS[@]}" coreos-devel/sdk-depends world' exited with nonzero code: 1
```

Is there anything to do other than tell people to download a new sdk?

Reply:
Is this with the updated ebuilds? gobject-introspection is at 1.68 in your PR.

But otherwise yeah, "download new sdk" seems like the only solution.

Reply:
Oh, right, I forgot to update the repos. Trying again then.

Reply:
I'll try again next week on some different SDK - current one failed because it wanted to rebuild rust, because the python3 sdk still had the old version.

Reply:
So update_chroot worked, but I had to do a small `profiles/updates` update in portage-stable for `dev-util/dwarves` rename into `dev-util/pahole`. Python also seems to work just fine, without the need for `eselect python update` even. `eselect python show` shows `python3.9`, so it's all good.